### ISSUE TITLE
What should happen when an activity has both 2 countries AND 2 regions with %'s defined?

### ISSUE THREAD

**Issue by ghost** [IATI/activity-details-split-by-fields#6]
A basic algorithm would then result in 4 split transactions:
* Country A, Region A
* Country A, Region B
* Country B, Region A
* Country B, Region B

But should we be taking into account anything about those regions? Maybe region A and B only exist in country A for instance? Is this bad input data? 

I'm sure I remember reading something about this in one of the other writeups but now I can't find it.

**Comment by odscjames** [comment]
See test `test_split_by_everything` which currently doesn't actually do that - but it shouldn't till this Q is resolved.

**Comment by Bjwebb** [comment]
Both the country and region percentages should sum together to make 100%.

https://iatistandard.org/en/iati-standard/203/activity-standard/iati-activities/iati-activity/recipient-country/#example-usage

> When both the recipient-region and recipient-country are declared, then the @percentage values should sum to 100% per region vocabulary for the specific iati-activity.

```
<recipient-country code="AF" percentage="50" />
<recipient-region code="489" vocabulary="1" percentage="50" />
<recipient-region code="A1" vocabulary="99" percentage="50" vocabulary-uri="http://example.com/vocab.html" />
```

**Comment by ghost** [comment]
Ah, I had heard that but didn't understand the significance.

So, when an activity has both 2 countries AND 2 regions with %'s defined you should get:

*    Country A
*    Country B
*    Region A
*    Region B

And the sum value of those 4 transactions should be the same as the value on the original transaction?

**Comment by Bjwebb** [comment]
That's right. But it gets more complicated when there's regions from multiple vocabularies.

**Comment by ghost** [comment]
Ok, how should it work in that case?

Activity has 2 countries (A and B) and 2 regions in one vocab (C and D) and 2 regions in another (E and F), the split on each is 25%.

Should you get?
* Country A, 25%
* Country B, 25%
* Region C, 25%
* Region D, 25%
* Region E, 25%
* Region F, 25%

So now adding up all the values of all these transactions gets you more than the original value! But if you add up per vocab (and including anything not marked for a region vocab) you get:
* A + B + C + D = 100%
* A + B + E + F = 100%

[ Except the adding up gets more complicated as sectors can also have multiple vocabs ]

**Comment by Bjwebb** [comment]
Yes, that's right.
